### DISCUSSION TITLE
Installing broke my server’s internet connectivity.

### DISCUSSION THREAD

**Discussion by MaximalCats** [angristan/wireguard-install#398] [Help]
Installing wireguard on debian 11 broke my server’s internet.
Edit: removed wireguard, didnt fix it.
`ping: google.com: Temporary failure in name resolution `

**Comment by ADMrugacz** [comment]
Same here. Look at the resolv.conf file and make sure you have nameservers listed in there. Then restart it. My copy had the nameservers removed after removing WireGuard and was masked and I had to unmask it to restart it. This is on Ubuntu LTS 22.04  system.

**Comment by MSe1969** [comment]
> Installing wireguard on debian 11 broke my server’s internet. Edit: removed wireguard, didnt fix it. `ping: google.com: Temporary failure in name resolution `

Wireguard needs IP-Forwarding, which is therefore enabled in the **/etc/sysctl.conf** by the 'wireguard-install' script - when enabling IP6-forwarding, the so-called "router advertisement" functionality is switched off - this is also described in the respective comment section:
```
# Uncomment the next line to enable packet forwarding for IPv6
#  Enabling this option disables Stateless Address Autoconfiguration
#  based on Router Advertisements for this host
#net.ipv6.conf.all.forwarding=1
```
The script in fact adds the above setting. Depending on the VPS hosting provider's specific network setup, this breaks in some cases the IPv6 functionality. On top, in one case, which I have experienced on a VPS hosted at IONOS, their **/etc/network/interfaces** file on Debian 11 for their standard configuration looks as follows:

```
iface ens192 inet6 manual
      pre-up sleep 5
      up dhclient -6 -nw -v ens192
      down dhclient -6 -r -v ens192
```
What happens here is, that router advertisement is used to trigger DHCP and the DHCP client is forced to act, even if no IPv6 address has been acquired yet (-nw parameter) - as a consequence in that case, the file **/etc/resolv.conf** was rewritten with IPv6 DNS resolver addresses - which corresponds in combination with a non-working IPv6 functionality symptoms indicating a loss of internet connectivity. To get IPv4 resolvers back, simply execute  `dhclient -4`

On my IONOS box, I have solved the issue by replacing the IPv6 config in the /etc/network/interfaces as follows by working with a static IP address (replace the IPv6 address place holder with your real one!):
```
iface ens192 inet6 static
        accept_ra 2
        address 2a00:xxxx:xxxx:67::1/128
        up ip -6 r a default dev ens192
```
DON'T blindly try to adopt this solution, if you don't have a IONOS VPS - it really depends on the specific IPv6 auto-configuration in your VPS - playing around with this may even cause your system not to boot any more, so be prepared to get into rescue mode.
Later, I have learned about another option worth a try, too: In above situation, the interface is "ens192" so you could instead also try adding the value "net.ipv6.conf.**ens192**.accept_ra=2" to the file /etc/sysctl.conf and executing `sudo sysctl -p` (use **your** interface value accordingly as highlighted in bold) - after repair, reboot your instance.

**Comment by Nolux** [comment]
I am having the same issues. Tried uninstalling with the script to revert the changes, but it still happens on every restart.

**Comment by x-FBi** [comment]
The Problem looks like your DNS.  Fix the DNS
[https://www.tecmint.com/set-permanent-dns-nameservers-in-ubuntu-debian/](url)

**Reply by MaximalCats** [reply]
Haha, thanks for your reply but this issue is from a year ago lol. I'll probably close this issue

**Reply by wepoo** [reply]
> Haha, thanks for your reply but this issue is from a year ago lol. I'll probably close this issue

So did you just reinstall your OS?

**Comment by Merlincool** [comment]
No there is no solution, it's not working on Debian 10 vps and it breaks internet connectivity. Please give some solution on this.

**Reply by x-FBi** [reply]
The answer was to fix your DNS
